### DISCUSSION TITLE
MultiChannel

### DISCUSSION THREAD

**Discussion by oscarramirezs** [Project-MONAI/MONAI#2017] [Q&A]
hello all
can someone help me please.

- I have few CT scan to make a segmentation model,  I would like to segment two bones with this model. 

- the problem is,  I have a file .nii for the CT, one file .nii for bone  #1 and one file .nii for bone #2.

- is there a tutorial to convert these two files into a single multichanel file? or what solusion can I use, 
-
- I want to use the Brain tumor 3D segmentation example for the segmentation

thank you 
OB

**Comment by rijobro** [comment]
Are the images of the same dimension? If yes, you can simply read them and then concatenate the arrays. If not, you can use `Orientationd` (to make sure they're similarly orientated) and `Spacingd` (to ensure they have the same voxel spacing).

**Comment by oscarramirezs** [comment]
Hi @rijobro thank you for your help
 the images are the same size 512x512 but different deep ex ( 512, 512, 200) and (512, 512, 350)
  -I have ( train images  - train_labels_humerus -train_labels_scapula)
 
```python
[{'image': 'Images_256/DEAB621C050A_A_mimics.nii',
  'label': {'label_scapula': 'Masks_scapula_256/DEAB621C050A_A_mimics_scapula.nii',
   'label_humerus': 'Masks_humerus_256/DEAB621C050A_A_mimics_humerus.nii'}}]
```


I did the concanate and I have the label in to channels ex (2 , 512, 512, 200)

 ```python
loader = LoadImaged(keys=("label_scapula", "label_humerus"))
for i in range(len(data_dict_l)):
    data_dict = loader(data_dict_l[i])
    label_scapula, label_humerus = data_dict["label_scapula"], data_dict["label_humerus"]
    add_channel = AddChanneld(keys=["label_scapula", "label_humerus"])
    datac_dict = add_channel(data_dict)
    mask= np.concatenate((datac_dict['label_scapula'], datac_dict['label_humerus']), axis=0)
    label_bones.append(mask)
    i+=1
label_bones[0].shape
```
(2, 512, 512, 256)


**### > but` I'm tryin do doing into the train_transforms = Compose(..**

using the tutorial (Brain tumor 3D segmentation with MONAI), it crashes 
 Do you have an example of a class that I can use for this? I will appreciate it very much
  
 

> like   

```python
class ConvertToMultiChannelBasedOnBratsClassesd(MapTransform):
    """

    """

    def __call__(self, data):
        d = dict(data)
        for key in self.keys:
            result = []
            # merge label 2 and label 3 to construct TC
            result.append(np.logical_or(d[key] == 2, d[key] == 3))
            # merge labels 1, 2 and 3 to construct WT
            result.append(
                np.logical_or(
                    np.logical_or(d[key] == 2, d[key] == 3), d[key] == 1
                )
            )
            # label 2 is ET
            result.append(d[key] == 2)
            d[key] = np.stack(result, axis=0).astype(np.float32)
        return d
```


thanks Oscar

**Reply by rijobro** [reply]
You're going to have to be more specific about the error you're getting. Also: 

- if you concatenate an image of size `512, 512, 200` with an image of size `512, 512, 350`, that will cause an error.
- you have an `i+=1` which is unnecessary inside of a for loop.
